lib/diagnostic.js

```diff
@@ -6,7 +6,11 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
-
+  let minvalue = 0
+  for (let i = 1; i<Array.length; i++)
```

Comment:
This would definitely work, but since this is a function that is actually going to be passed to reduce function there is no need for a loop.